packages/visualizations/src/components/utils/formatter.ts

```diff
@@ -22,7 +22,7 @@ export function defaultCompactLegendNumberFormat(value: number): string {
 }
 
 export function assureMaxLength(value: string, maxLength: number) {
-    if (value.length > maxLength) {
+    if (value && value.length > maxLength) {
```

Comment:
```suggestion
    if (value?.length > maxLength) {
```

Reply:
I'm surprised we don't have a rule for that 🤔

Reply:
You're right about the suggestion in this case but I have the feeling that such rule will be hard to enforce in all situations, in some cases optional chaining might not be recommended as optional chaining and AND operator don't give always the same output ?